pulp_smash/pulp3/bindings.py

```diff
@@ -0,0 +1,57 @@
+from unittest import TestCase
+from time import sleep
+
+from pulpcore.client.pulpcore import ApiClient, TasksApi
+
+from pulp_smash.api import _get_sleep_time
+from pulp_smash.config import get_config
+
+
+cfg = get_config()
+SLEEP_TIME = _get_sleep_time(cfg)
+configuration = cfg.get_bindings_config()
+pulpcore_client = ApiClient(configuration)
+tasks = TasksApi(pulpcore_client)
+
+
+def monitor_task(task_href):
+    """Polls the Task API until the task is in a completed state.
+
+    Prints the task details and a success or failure message. Exits on failure.
+
+    Args:
+        task_href(str): The href of the task to monitor
+
+    Returns:
+        list[str]: List of hrefs that identify resource created by the task
+
+    """
+    completed = ["completed", "failed", "canceled"]
+    task = tasks.read(task_href)
+    while task.state not in completed:
+        sleep(SLEEP_TIME)
+        task = tasks.read(task_href)
+
+    if task.state == "completed":
+        return task.created_resources
+
+    return task.to_dict()
```

Comment:
I would prefer if failed task threw an exception. It would make for easier tests and the function would have a single return type.